crates/solvers/src/boundary/naive.rs

```diff
@@ -0,0 +1,245 @@
+use {
+    crate::{
+        boundary::liquidity::constant_product::to_boundary_pool,
+        domain::{eth, liquidity, order, solution},
+    },
+    ethereum_types::{H160, U256},
+    itertools::Itertools,
+    model::order::{
+        LimitOrderClass,
+        Order,
+        OrderClass,
+        OrderData,
+        OrderKind,
+        OrderMetadata,
+        OrderUid,
+    },
+    num::{BigRational, One},
+    solver::{
+        liquidity::{
+            slippage::{SlippageCalculator, SlippageContext},
+            AmmOrderExecution,
+            ConstantProductOrder,
+            Exchange,
+            LimitOrder,
+            LimitOrderId,
+            LiquidityOrderId,
+            SettlementHandling,
+        },
+        settlement::{external_prices::ExternalPrices, SettlementEncoder},
+        solver::naive_solver::multi_order_solver,
+    },
+    std::sync::{Arc, Mutex},
+};
+
+pub fn solve(
+    orders: &[&order::Order],
+    liquidity: &liquidity::Liquidity,
+) -> Option<solution::Solution> {
+    let pool = match &liquidity.state {
+        liquidity::State::ConstantProduct(pool) => pool,
+        _ => return None,
+    };
+
+    // Note that the `order::Order` -> `boundary::LimitOrder` mapping here is
+    // not exact. Among other things, the signature and various signed order
+    // fields are missing from the `order::Order` data that have access to in
+    // the solver engines. This means that the naive solver in the `solver`
```

Comment:
```suggestion
    // the solver engines. This means that the naive solver in the `solvers`
```

Reply:
I meant it as `solver::naive_solver` in that it would produce invalid `Settlement` values, but that `solvers::naive_solver` doesn't care because it can extract the information it needs anyway.